kernel/syscall.c

```diff
@@ -1881,6 +1881,95 @@ long myst_syscall_fstatfs(int fd, struct statfs* buf)
     return ret;
 }
 
+long myst_syscall_fchownat(
+    int dirfd,
+    const char* pathname,
+    uid_t owner,
+    gid_t group,
+    int flags)
+{
+    long ret = 0;
+    struct locals
+    {
+        char realpath[PATH_MAX];
+        char abspath[PATH_MAX];
+    };
+    struct locals* locals = NULL;
+
+    if (!pathname)
+        ERAISE(-EINVAL);
+
+    if ((flags & ~(AT_EMPTY_PATH | AT_SYMLINK_NOFOLLOW)) != 0)
+        ERAISE(-EINVAL);
+
+    if (!(locals = malloc(sizeof(struct locals))))
+        ERAISE(-ENOMEM);
+
+    /* If pathname is absolute, then ignore dirfd */
+    if (*pathname == '/' || dirfd == AT_FDCWD)
```

Comment:
I think all these cases can be consolidated with:
```c
   helper_function_to_get_real_path(dirfd, pathname, &locals->realpath);
   ret = (flags & AT_SYMLINK_NOFOLLOW) ? 
                myst_syscall_lchown(locals->realpath, owner, group) : 
                myst_syscall_chown(locals->realpath, owner, group);
```

Reply:
Thanks for the suggestion, have incorporated it.